iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/XMLExporter.java

```diff
@@ -89,16 +106,14 @@ public void export(Network network, Properties parameters, DataSource dataSource
             throw new IllegalArgumentException("network is null");
         }
 
-        ExportOptions options = new ExportOptions();
-        if (parameters != null) {
-            options.setIndent(Boolean.parseBoolean(parameters.getProperty(INDENT_PROPERTY, Boolean.TRUE.toString())))
-                .setWithBranchSV(Boolean.parseBoolean(parameters.getProperty(WITH_BRANCH_STATE_VARIABLES_PROPERTY, Boolean.TRUE.toString())))
-                .setOnlyMainCc(Boolean.parseBoolean(parameters.getProperty(ONLY_MAIN_CC_PROPERTIES, Boolean.FALSE.toString())))
-                .setAnonymized(Boolean.parseBoolean(parameters.getProperty(ANONYMISED_PROPERTIES, Boolean.FALSE.toString())))
-                .setSkipExtensions(Boolean.parseBoolean(parameters.getProperty(SKIP_EXTENSIONS_PROPERTIES, Boolean.FALSE.toString())))
-                .setTopologyLevel(TopologyLevel.valueOf(parameters.getProperty(TOPOLOGY_LEVEL_PROPERTY, TopologyLevel.NODE_BREAKER.name())))
-                .setThrowExceptionIfExtensionNotFound(Boolean.parseBoolean(parameters.getProperty(THROW_EXCEPTION_IF_EXTENSION_NOT_FOUND, Boolean.TRUE.toString())));
-        }
+        ExportOptions options = new ExportOptions()
+                .setIndent((boolean) Converters.readParameter(getFormat(), parameters, INDENT_PARAMETER, defaultValueConfig))
+                .setWithBranchSV((boolean) Converters.readParameter(getFormat(), parameters, WITH_BRANCH_STATE_VARIABLES_PARAMETER, defaultValueConfig))
+                .setOnlyMainCc((boolean) Converters.readParameter(getFormat(), parameters, ONLY_MAIN_CC_PARAMETER, defaultValueConfig))
+                .setAnonymized((boolean) Converters.readParameter(getFormat(), parameters, ANONYMISED_PARAMETER, defaultValueConfig))
+                .setSkipExtensions((boolean) Converters.readParameter(getFormat(), parameters, SKIP_EXTENSIONS_PARAMETER, defaultValueConfig))
+                .setTopologyLevel(TopologyLevel.valueOf((String) Converters.readParameter(getFormat(), parameters, TOPOLOGY_LEVEL_PARAMETER, defaultValueConfig)))
+                .setThrowExceptionIfExtensionNotFound((boolean) Converters.readParameter(getFormat(), parameters, THROW_EXCEPTION_IF_EXTENSION_NOT_FOUND_PARAMETER, defaultValueConfig));
```

Comment:
Could you try to add convenient method in Converters class to return the expected type?